### ISSUE TITLE
Problems with RouteStrategy in Blazor server-side project

### ISSUE THREAD

**Issue by sergeyborysenko** [Finbuckle/Finbuckle.MultiTenant#906]
Hi,
I am currently working on implementing multitenancy in our Blazor project authenticating towards Keycloak.
Initially I have chosen Route strategy, as we have hundreds of tenants pointing to the same hosted app instance, and then it seems the easiest way to support without additional configuration, in contrary to DNS or base path strategies.

And all the configuration worked fine, except the problem with Blazor's specific. Blazor has built-in SignalR functionality with constant client-server communication for updates.

And when I open a page and when through authentication pipeline after 1-2 seconds Blazor sends a request to something like 
`CONNECT https://localhost:7221/_blazor?id=uWPEjzY3ntFknkIl3sT1ow HTTP/2`
And this route leads to not resolving a tenant and correspondently changes the current user to a non-authenticated state, that of course leads in my Routes.razor to the Unauthorised page.

Maybe you can advice how to deal with this situation?

**Comment by sergeyborysenko** [comment]
Just wanted to add that currently I have found a workaround to bypass the issue adding a middleware that basically does this simple thing:
```
 if (httpContext.Request.Path.Equals("/_blazor"))
 {
     httpContext.Items.Add("__tenant____bypass_validate_principal__", true);
 }
```
And it helps. However this method feels a bit of a hack and I am not sure if it can't be used to bypass my authorization in the app. 
And also as those blazor SignalR request come last, my navigation links become broken as they combine the routes base on the tenant value, i.e .  `Href=@($"{TenantAccessor.MultiTenantContext?.TenantInfo?.Identifier}/users") `
Of course, I can also find a workaround for that as well.

**Comment by AndrewTriesToCode** [comment]
Hi, I see what you mean. Have you found a better solution yet? Blazor presents some interesting challenges.

I have had success by have a custom strategy that is registered before the route strategy that looks to see if the `_blazor` endpoint is being hit and if so will try to get the current tenant from the `__tenant__` property of the authentication result which can be accessed with the `IAuthenticationResultFeature`. The library encodes the tenant at sign in time into this property. If the endpoint isn't `_blazor` then the strategy just returns null and the next strategy (route in your case) is tried.

**Comment by sergeyborysenko** [comment]
To be honest, at some point, I gave up on the route strategy and pivoted to the host strategy—and it worked flawlessly without a single issue. From a configuration perspective, it wasn’t as complex as I initially feared. A simple wildcard DNS redirection to the same host was all it took.

With the route strategy, every time I found a workaround for one issue, another would crop up. It required combining the route strategy with a custom strategy, implementing a specific tenant-resolving wrapper for my Blazor Routes component, creating a dedicated resolver for the navigation menu, and even then, I encountered issues with cookies and the login/logout flow.

Maintaining that approach felt overly complicated and demanded every developer on the team to fully understand its intricacies. In contrast, the host strategy worked as intended right out of the box and has been much easier to maintain.
